### PR TITLE
Update stats

### PR DESCRIPTION
**Pull request by nlehuby**
patch the cli behaviour and update the stats with the count by zone_type

### CODE REVIEW

File: src/cosmogony.rs [hunk 1]
```diff
@@ -35,6 +37,15 @@ impl CosmogonyStats {
             }
         });
     }
+    pub fn process_zone_types(&mut self, zones: &Vec<Zone>) {
+        for zone in zones {
+            let type_ = zone.zone_type
+                .unwrap_or(ZoneType::NonAdministrative)
```
Comment (nlehuby) [inline src/cosmogony.rs]: I'm not very proud of this. Any idea to deal with it in a better way ? :thinking:

Comment (antoine-de) [reply src/cosmogony.rs]: and if you just change `zone_type_counts` to be `BTreeMap<Option<ZoneType>, u64>` ?

this way we would have stats on unknown zone_types no ?

File: src/lib.rs [hunk 2]
```diff
@@ -64,7 +64,7 @@ pub fn get_zones_and_stats(
             if let Some(zone) = zone::Zone::from_osm_with_geom(relation, &objects, next_index) {
                 // Ignore zone without boundary polygon for the moment
                 if zone.boundary.is_some() {
-                    stats.process(&zone);
+                    stats.init(&zone);
```
Comment (antoine-de) [inline src/lib.rs]: why call it init ?

File: src/cosmogony.rs [hunk 3]
```diff
@@ -35,6 +37,15 @@ impl CosmogonyStats {
             }
         });
     }
+    pub fn process_zone_types(&mut self, zones: &Vec<Zone>) {
```
Comment (antoine-de) [inline src/cosmogony.rs]: can't we do both `init` and `process_zone_types` at the same time (doing it later then)

File: src/lib.rs (hunk 2 above)
Comment (nlehuby) [reply src/lib.rs]: any better name idea ? `process` seems to generic, but I agree, `init` is not really better

File: src/cosmogony.rs (hunk 3 above)
Comment (nlehuby) [reply src/cosmogony.rs]: :+1:

File: src/lib.rs [hunk 4]
```diff
@@ -53,7 +53,7 @@ pub fn get_zones_and_stats(
     info!("reading pbf done.");
 
     let mut zones = vec![];
-    let mut stats = CosmogonyStats::default();
```
Comment (antoine-de) [inline src/lib.rs]: this stat obj is still used ?

File: src/lib.rs [hunk 5]
```diff
@@ -80,7 +79,7 @@ pub fn get_zones_and_stats_without_geom(
     info!("Reading pbf without geometries...");
 
     let mut zones = vec![];
-    let mut stats = CosmogonyStats::default();
```
Comment (antoine-de) [inline src/lib.rs]: same here, I think you can remove it

File: src/lib.rs (hunk 4 above)
Comment (nlehuby) [reply src/lib.rs]: yes, it is returned by the current function (`get_zones_and_stats`)

**Pull request merged**